### PR TITLE
tests[CL]: Add SwapInGivenOut Inverse Test

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #4164

## What is the purpose of the change

Extending on #4076, we want to test the same inverse tests that we did for SwapOutGivenIn, but this time for SwapInGivenOut. This ensures we preserve the amount swapped when we do a swap from `denom1 -> denom2` and then back from `denom2 -> denom1`.

Note to reviewer: The diff is rendering a little strange.

## Brief Changelog

- Adds `TestInverseRelationshipSwapInAmtGivenOut`
- Creates a shared `inverseRelationshipInvariants` helper function
- Refactors `TestInverseRelationshipSwapOutAmtGivenIn` to use the helper function

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Review by mattverse — APPROVED** [review]
I love how you approached it by creating a method for the inverse invariants! LGTM 🌮

**Comment by czarcas7ic** [comment]
Thanks Matt! @p0mvn please lmk if I should merge this into your branch or if you want me to wait for yours to be merged and then I will just merge to main.

**Review by p0mvn — APPROVED** [review]

**Pull request merged**
